### PR TITLE
QDB-2594 - Improve teamcity integration.

### PR DESCRIPTION
**Pull request by vianneyPL**
This is a two phase operation.

The first is to modify the teamcity configuration to rely on the vcs root `teamcity-qdb-test-setup` to add the scripts used to setup qdbd, and add the start-services and stop-services step in the build phase.

The second one is to remove what was done to setup qdbd in the golang test framework.

### CODE REVIEW

**Comment by vianneyPL** [comment]
You can see it done there:
https://teamcity.quasardb.net/buildConfiguration/Api_Go_LinuxWithBuildMultistep?branch=qdb-2594-improve-teamcity-integration&buildTypeTab=overview

File: all_test.go [hunk 1]
```diff
@@ -11,79 +11,21 @@ import (
...
-	usersConfigFile       string = "users.cfg"
-	qdbPort               int    = 30083
-	nodeURI               string = "qdb://127.0.0.1:30083"
-	clusterURI            string = "qdb://127.0.0.1:30083"
-	securedURI            string = "qdb://127.0.0.1:30084"
+	clusterPublicKeyFile string = "cluster_public.key"
+	userPrivateKeyFile   string = "user_private.key"
+	usersConfigFile      string = "users.cfg"
+
+	clusterURI string = "qdb://127.0.0.1:2836"
```
Comment (mkurdej) [inline all_test.go]: Please rename (everwhere / as much as possible): clusterURI -> insecureURI, secured -> secure to make it explicit which cluster is secure and which is insecure.

File: cluster_test.go [hunk 2]
```diff
@@ -60,32 +60,15 @@ var _ = Describe("Tests", func() {
...
-				var nodeStat Statistics
-				for _, stat := range stats {
-					nodeStat = stat
-					break
-				}
-				Expect(int64(0)).ToNot(BeNumerically("==", nodeStat.Persistence.EntriesCount))
-
-				err = cluster.PurgeCache()
-				Expect(err).ToNot(HaveOccurred())
+				// we cannot really test that
```
Comment (mkurdej) [inline cluster_test.go]: Why?

File: entry_test.go [hunk 3]
```diff
@@ -267,7 +267,7 @@ var _ = Describe("Tests", func() {
 			It("should locate", func() {
 				location, err := integer.GetLocation()
 				Expect(err).ToNot(HaveOccurred())
-				Expect(int16(qdbPort)).To(Equal(location.Port))
+				Expect(int16(2836)).To(Equal(location.Port))
```
Comment (mkurdej) [inline entry_test.go]: So here you suppose insecure cluster. Why can't you make it a parameter/variable?

File: node_test.go [hunk 4]
```diff
@@ -14,7 +14,7 @@ var _ = Describe("Tests", func() {
 	)
 
 	BeforeEach(func() {
-		handle, err = SetupHandle("qdb://127.0.0.1:30083", 120*time.Second)
+		handle, err = SetupHandle("qdb://127.0.0.1:2836", 120*time.Second)
```
Comment (mkurdej) [inline node_test.go]: Why don't you use the above defined URIs?

File: cluster_test.go (hunk 2 above)
Comment (vianneyPL) [reply cluster_test.go]: I added a test that test that :)

File: all_test.go (hunk 1 above)
Comment (vianneyPL) [reply all_test.go]: Done.

File: entry_test.go (hunk 3 above)
Comment (vianneyPL) [reply entry_test.go]: I don't like the idea of creating a global variable for only one place in the code.

Comment (mkurdej) [reply entry_test.go]: Do you prefer magic numbers then?

File: node_test.go (hunk 4 above)
Comment (vianneyPL) [reply node_test.go]: Done.

File: cluster_test.go [hunk 5]
```diff
@@ -61,31 +61,28 @@ var _ = Describe("Tests", func() {
...
 				Expect(err).ToNot(HaveOccurred())
-				var nodeStat Statistics
+				Expect(len(stats)).To(BeNumerically(">", 0))
+
+				var memBefore int64
 				for _, stat := range stats {
-					nodeStat = stat
+					memBefore = stat.Memory.BytesResident
 					break
 				}
```
Comment (terngkub) [inline cluster_test.go]: Why do you use for loop here and not access the elem directly?

Comment (terngkub) [reply cluster_test.go]: I was thinking something like
`
if len(stats) > 0 {
    memBefore = stats[0].Memory.BytesResident
}
`

**Pull request closed without merging**